compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/overrides/IrVisibilityUtil.kt

```diff
@@ -16,13 +15,12 @@ fun isVisibleForOverride(overriding: IrOverridableMember, fromSuper: IrOverridab
 }
 
 fun IrDeclarationWithVisibility.isEffectivelyPrivate(): Boolean {
-    fun DescriptorVisibility.isNonPrivate(): Boolean =
-        this == DescriptorVisibilities.PUBLIC
-                || this == DescriptorVisibilities.PROTECTED
-                || this == DescriptorVisibilities.INTERNAL
+    val isNonPrivate = visibility == DescriptorVisibilities.PUBLIC
+            || visibility == DescriptorVisibilities.PROTECTED
+            || visibility == DescriptorVisibilities.INTERNAL
 
     return when {
-        visibility.isNonPrivate() -> parentClassOrNull?.isEffectivelyPrivate() ?: false
+        isNonPrivate -> parentClassOrNull?.isEffectivelyPrivate() ?: false
```

Comment:
My god, these double negations with fallbacks to `null == false` are quite bad. Just why…

Reply:
It ain't owe nothing to nobody!